go-controller/pkg/ovn/egressip.go

```diff
@@ -1466,7 +1466,8 @@ func (oc *DefaultNetworkController) syncStaleSNATRules(egressIPCache map[string]
 			klog.Infof("syncStaleSNATRules will delete %s due to logical ip: %v", egressIPName, item)
 			return true
 		}
-		if !cacheEntry.egressIPs.Has(item.ExternalIP) {
+		if node, ok := cacheEntry.egressIPs[item.ExternalIP]; !ok ||
+			item.LogicalPort == nil || *item.LogicalPort != types.K8sPrefix+node {
 			klog.Infof("syncStaleSNATRules will delete %s due to external ip: %v", egressIPName, item)
```

Comment:
nit: let's add logical port in the message too, please:
`syncStaleSNATRules will delete %s due to external ip or stale logical port: %v"...`